Ported to python3

benhg opened the pull request.

Commit:
Ported to python3

Commit:
Merged intelligent-agent/dev into our python3Port branch

benhg commented:
These are failing because Travis is set to run with Python2.7, but obviously due to the nature of our changes (involving python3), it will fail. We're going to make travis run the integration tests on Py3.

benhg commented:
I think this has been fixed by the wonderful @roshbhatia. The tests should all pass now.

eliasbakken commented on scripts/.#uninstall.sh:
```diff
@@ -0,0 +1 @@
+toggle@osboxes.4290:1551218152
```
Is this necessary?

ThatWileyGuy replied:
And why are there three of them?
And why does the only one with a meaningful command in it delete the entire OS? :D

ThatWileyGuy commented on toggle/CascadingConfigParser.py:
```diff
@@ -18,15 +18,15 @@
 
 
-class CascadingConfigParser(ConfigParser.SafeConfigParser):
+class CascadingConfigParser(configparser.SafeConfigParser):
```
Are you seeing any warnings that SafeConfigParser is deprecated?
I saw them when I started poking at 2to3 in Redeem, but if this works without warnings we can address it later?

ThatWileyGuy commented on toggle/jogtest.py:
```diff
@@ -0,0 +1,3 @@
+from Jog import Jog
+
+print("Hello")
```
And what's up with this?

ThatWileyGuy commented on toggle/pkg_resources:
```diff
@@ -0,0 +1,610 @@
+# This file helps to compute a version number in source trees obtained from
```
Does this need to be checked in or does Versioneer generate this on demand?

If this does need to be checked it, can it be checked in separately from all the 2to3 changes?

ThatWileyGuy replied:
It looks like you added and then removed Versioneer and this was left over?

ThatWileyGuy commented:
In general, this is fine. There are a few mysterious changes that should be explained or removed, and I'll sign off.

It would also be great to squash these commits so you're not adding and later removing files - check out `git rebase -i HEAD~6` and asking git to squash all your commits into the first one. You'll have to force-push to update the PR, but it should make things a lot cleaner.

Commit:
Making Requested Changes - Deleted extraneous files

Updated travis.yml

Reformatted _version.py(in build/) and scale_ui.py

remove accidentally committed build files in commit 5a86832eea38003ee67fa2f409a063fb1c07affa


making requested changes

roshbhatia commented:
@ThatWileyGuy Ben and I finished + squashed the changes, can you please check to see if we did ok?

ThatWileyGuy commented:
Looks good to me, and I think Github will automagically squash away that merge commit for you.

Wackerbarth commented:
Did anyone actually try this on real hardware before the merge? I get a failure to start the toggle service.

goeland86 commented:
I was having difficulties on my BBBW to get the Clutter backend to load - but didn't get it working on HW yet. The fact that it works in the VM is already a good indication the issue lies in underlying libraries rather than in the code itself though.

Wackerbarth commented:
Yes, but, until it actually works, I think that it needs to be kept as a "proposed feature" rather than being merged into the master "dev" branch.

goeland86 commented:
I would argue that we should repair it directly in the `dev` branch now if there is anything to fix, and Umikaze RC3 will ship with the py3 version of toggle. I'm trying to build it this weekend.

Wackerbarth commented:
That is where we have a serious difference of opinion with respect to workflow. You seem to dump in things that "probably will work" and make everyone work around the failures. OTOH, I don't want to merge in anything until it has been demonstrated to work.

As to Umikaze RC3, I would argue that we should have released a python2 version of Toggle 1.3 and already created that candidate. Instead, you seem to be willing to continue to delay things even more.
At this rate, we will never actually have a release.

ThatWileyGuy commented:
Sorry, for some reason I had it in my head that the students had actual hardware. I'll take a look at what's broken.

goeland86 commented:
They do have hardware - but they're developing in parallel tracks between
on-hardware and VM, to advance faster.

On Thu, Mar 21, 2019 at 3:34 PM Andrew Wiley <notifications@github.com>
wrote:

> Sorry, for some reason I had it in my head that the students had actual
> hardware. I'll take a look at what's broken.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/intelligent-agent/toggle/pull/24#issuecomment-475254217>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AQiOefHtzpblZPtsT0iIIXxgc3NYnDcUks5vY5htgaJpZM4b1PFQ>
> .
>

ThatWileyGuy commented:
Ah, okay.

The only thing broken is that pyconnman is not compatible with Python 3. You have to comment out the relevant lines in get_manager() in toggle/Network.py so we don't try to import it.

Aside from that...
- delete and recreate the virtualenv with `python -m virtualenv -p python3 --system-site-packages /usr/local/Toggle-develop`
- `python setup.py develop`
- `apt install python3-gi python3-gi-cairo python3-numpy python3-tornado python3-networkmanager`

ThatWileyGuy commented:
Looks like someone opened a PR to fix pyconnman, but the project has been dormant for 5 years and it's not clear if it'll get merged: https://github.com/liamw9534/pyconnman/pull/4

goeland86 commented:
In any case we don't have connman installed as part of Umikaze normally -
if we do we should actually remove it from the base image, as it often
conflicts with NetworkManager anyway.

On Thu, Mar 21, 2019 at 3:57 PM Andrew Wiley <notifications@github.com>
wrote:

> Looks like someone opened a PR to fix pyconnman, but the project has been
> dormant for 5 years and it's not clear if it'll get merged:
> liamw9534/pyconnman#4 <https://github.com/liamw9534/pyconnman/pull/4>
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/intelligent-agent/toggle/pull/24#issuecomment-475263800>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AQiOeSBya0GVxiD-g5zFzr7QcXhjFKNMks5vY53egaJpZM4b1PFQ>
> .
>
